letsencrypt-win-simple/Scripts/ImportExchange.ps1

```diff
@@ -1,56 +1,95 @@
 param(
     [Parameter(Position=0,Mandatory=$true)]
-    [string]$NewCertThumbprint,
+    [string]$NewCertThumbprint,         # should be {5} from letsencrypt.exe
     [Parameter(Position=1,Mandatory=$true)]
-    [string]$ExchangeServices
+    [string]$ExchangeServices,          # comma-separated list of exchange services to import for
+    [Parameter(Position=2,Mandatory=$false)]
+    [bool]$LeaveOldExchangeCerts = 1,   # 0 for false, 1 for true. If set to 1, will skip removal of old exchange certs
+    [Parameter(Position=3,Mandatory=$false)]
+    [string]$TargetHost,                # primary fqdn of cert. shouldn't be necessary if thumbprint was included and letsencrypt places cert in the store
+    [Parameter(Position=4,Mandatory=$false)]
+    [string]$StorePath                  # required if $TargetHost specified. Cert pfx file will be stored here.
 )
 
+Write-Debug -Message ('NewCertThumbprint: {0}' -f $NewCertThumbprint)
+Write-Debug -Message ('ExchangeServices: {0}' -f $ExchangeServices)
+Write-Debug -Message ('TargetHost: {0}' -f $TargetHost)
+Write-Debug -Message ('StorePath: {0}' -f $StorePath)
+
+
 ## Imports new cert thumbprint into the Exchange roles defined in the $ExchangeServices comma-separate list
 ## example: $ExchangeServices = IMAP,POP,IIS
 ## That will import the certificate for IMAP, POP, and IIS roles in exchange
 ## Valid services:  IMAP | POP | UM | IIS | SMTP | Federation | UMCallRouter
 
-## THIS SCRIPT IS INCOMPLETE AND UNTESTED
+## THIS SCRIPT IS INCOMPLETE AND *mostly* UNTESTED (some modifications have come in from people using it successfully)
 ## Documentation referenced from https://technet.microsoft.com/en-us/library/aa997231(v=exchg.160).aspx
 
 # Should work with exchange 2007 and higher
-Get-PSSnapin -registered | ? {$_.Name -match "Microsoft.Exchange.Management.PowerShell" -and ($_.Name -match "Admin" -or $_.Name -match "E2010" -or $_.Name -match "SnapIn")} | Add-PSSnapin -ErrorAction SilentlyContinue
+Get-PSSnapin -registered | Where-Object {$_.Name -match "Microsoft.Exchange.Management.PowerShell" -and ($_.Name -match "Admin" -or $_.Name -match "E2010" -or $_.Name -match "SnapIn")} | Add-PSSnapin -ErrorAction SilentlyContinue
 
 #$OldThumbprint = (Get-Item -Path RDS:\GatewayServer\SSLCertificate\Thumbprint).CurrentValue
 $CertInStore = Get-ChildItem -Path Cert:\LocalMachine -Recurse | Where-Object {$_.thumbprint -eq $NewCertThumbprint} | Sort-Object -Descending | Select-Object -f 1
```

Comment:
Does `Sort-Object -Descending` by default sort on the correct date field, so that the newest are on top?

Reply:
I'm honestly not sure why i kept using sort, since I'm not specifying the field to sort on and, more importantly, because we are filtering by thumbprint so we will never get certs from different days.